src/components/DelegateCard.tsx

```diff
@@ -23,18 +26,31 @@ export const DelegateCard = ({
   hasShareButton,
   hasDelegateButton = true,
 }: Props) => {
+  const [copied, setCopied] = useState<boolean>(false)
+  const { network } = useNetwork()
   const navigate = useNavigate()
   const { search } = useLocation()
+  const [copyLink, setCopyLink] = useState<string>(
+    `${window.location.host}/${network}/${sanitizeString(name)}`,
+  )
   const shouldHideLongDescription =
     !longDescription || longDescription === shortDescription
 
+  useEffect(() => {
+    if (copied) {
+      setTimeout(() => setCopied(false), 1000)
+    }
+  }, [copied])
+
+  useEffect(() => {
+    setCopyLink(`${window.location.host}/${network}/${sanitizeString(name)}`)
+  }, [name, network])
+
```

Comment:
`copied` and `setCopied` aren't used.
`copyLink` can be set directly in a `useMemo`
Let's go with `useCallbacks` for the others, to make sure all const are up to date.
```suggestion
  const { network } = useNetwork()
  const navigate = useNavigate()
  const { search } = useLocation()
  const copyLink = useMemo(() => `${window.location.host}/${network}/${sanitizeString(name)}`, [name, network])
  const shouldHideLongDescription =
    !longDescription || longDescription === shortDescription

  const onDelegate = useCallback(() => {
    navigate(`/delegate/${address}${search}`)
  },[address, navigate, search])

  const onCopy = useCallback(() => {
    navigator.clipboard.writeText(copyLink)
    toast.success('Copied to clipboard', {
      duration: 1000,
    })
  }, [copyLink])
```